tests/test_logic.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+import logic
+import model
+
+
+@pytest.mark.parametrize("task_breadcrumbs,task_id,last_engine_decision_id,expected_message",
+                         [
+                             ([f"task_{n}" for n in range(1, 6)], "task_5", "task_4", ""),
+                             ([f"task_{n}" for n in range(1, 6)], "task_5", "task_5",
+                              "Are you sure you have done this task? Looks like you haven't."),
+                             ([f"task_{n}" for n in range(1, 6)], "task_5", "task_2",
+                              "You need to deal with a previous task."),
+                         ],
+                         ids=[
+                             "no message for next, consequitve task",
```

Comment:
I think you mean "consecutive"?

```suggestion
                             "no message for next consecutive task",
```